### PR TITLE
chore: few small css fixes and bug fixes

### PR DESCRIPTION
**Pull request by jac-ob**

### CODE REVIEW

**Comment by jac-ob** [comment]
Squash please.

File: lib/composites/menu/events/main.js [hunk 1]
```diff
@@ -14,25 +14,34 @@ const ACTIVE_CLASS = 'dqpl-active';
...
         !!closest(target, '.dqpl-menu-trigger');
+      const expandedDropdownItem = document.querySelector('[data-dropdown-expanded="true"]');
+      const menuExpanded = elements.menu.getAttribute('aria-expanded') === 'true';
+
+      if (!isWithinDropdown && expandedDropdownItem) {
+        expandedDropdownItem.click();
+      }
 
-      if (isWithin || isHamburger || isLocked) {
+      if (isWithinSideBar || isHamburger || isLocked) {
```
Comment (schne324) [inline lib/composites/menu/events/main.js]: to reduce the amount of `if` blocks, and to make this easier to read, I'd suggest:
```js
if (!isWithin && !isHamburder && !isLocked && menuExpanded) {
  onTriggerClick();
}
```

File: lib/composites/modals/style.less [hunk 2]
```diff
@@ -28,18 +28,28 @@
       border-top-left-radius: 3px;
       border-top-right-radius: 3px;
 
+      h1,
+      h2,
+      h3,
+      h4,
+      h5,
+      h6,
+      @{prefix}close {
```
Comment (schne324) [inline lib/composites/modals/style.less]: didn't we talk about a helper class for this?  you mind adding `@{prefix}header-item`

File: package.json [hunk 3]
```diff
@@ -67,5 +67,6 @@
     "simulant": "^0.2.2",
     "stringify": "^5.1.0",
     "vinyl-source-stream": "^1.1.0"
-  }
+  },
+  "dependencies": {}
```
Comment (schne324) [inline package.json]: just curious, was there a problem not having a `"dependencies"` key in package.json?

File: test/composites/menu/events/main.js [hunk 4]
```diff
@@ -73,7 +74,7 @@ describe('composites/menu/events/main', () => {
       setTimeout(() => {
         assert.equal(dropdown.getAttribute('aria-expanded'), 'true');
         done();
-      }, 400); // give animation/other timeouts a change to do stuff
```
Comment (schne324) [inline test/composites/menu/events/main.js]: problem with this comment?

File: lib/composites/menu/events/main.js [hunk 5]
```diff
@@ -14,25 +14,34 @@ const ACTIVE_CLASS = 'dqpl-active';
 
 module.exports = (elements) => {
   /**
-   * Listen for clicks outside the menu (when
-   * its opened AND not locked) to close it
+   * Listen for clicks outside the sidebar and topBar
```
Comment (schne324) [inline lib/composites/menu/events/main.js]: The open and locked bit doesn't apply to dropdowns. how about just:
```
/**
 * Listen for clicks outside sideBar and dropdowns within topBar
 */
```

File: lib/components/first-time-point/style.less [hunk 6]
```diff
@@ -5,15 +5,17 @@
...
   @{prefix}arrow-pointer {
-    width: 100px;
-    height: 60px;
+    width: 88px;
+    height: 37px;
     position: relative;
+    left: 6px;
     overflow: hidden;
-    box-shadow: 1px 25px 0 -22px @focus-glow, 0px 25px 6px -19px @focus-glow;
+    box-shadow: 0px 19px 0 -16px @focus-glow, 1px 18px 6px -12px @focus-glow;
```
Comment (schne324) [inline lib/components/first-time-point/style.less]: use the box-shadow [mixin](https://github.com/dequelabs/pattern-library/blob/master/lib/variables.less#L137) here and the one below on line 29

File: lib/components/first-time-point/style.less [hunk 7]
```diff
@@ -5,15 +5,17 @@
...
-    position: relative;
+    position: absolute;
+    top: -62px;
   }
 
   @{prefix}arrow-pointer {
-    width: 100px;
-    height: 60px;
+    width: 88px;
+    height: 37px;
```
Comment (schne324) [inline lib/components/first-time-point/style.less]: right on! does this reduce the footprint of the ftpo?

File: package.json (hunk 3 above)
Comment (jac-ob) [reply package.json]: I don't believe so, I never added it so an `npm i` must have

File: lib/components/first-time-point/style.less (hunk 7 above)
Comment (jac-ob) [reply lib/components/first-time-point/style.less]: Sure does!

File: lib/components/first-time-point/style.less (hunk 6 above)
Comment (jac-ob) [reply lib/components/first-time-point/style.less]: Did not think our mixin was setup for multiple, but I just figured out this fun little fact. In newer versions of less you can do `.box-shadow(0px 19px 0 -16px @focus-glow, 1px 18px 6px -12px @focus-glow;);` that extra semicolon is important. Will pick up on the comma separate as separate box-shadows and not args. Didn't know that. 👍

File: lib/composites/menu/events/main.js [hunk 8]
```diff
@@ -14,25 +14,29 @@ const ACTIVE_CLASS = 'dqpl-active';
...
+        expandedDropdownItem.click();
       }
 
-      if (elements.menu.getAttribute('aria-expanded') === 'true') {
+      if (!isWithinSideBar && !isHamburger && !isLocked && menuExpanded) {
         onTriggerClick();
       }
-    });
+
+    }, true);
```
Comment (schne324) [inline lib/composites/menu/events/main.js]: this concerns me...seems like it'd be easy to not have to `useCapture` here

Comment (jac-ob) [reply lib/composites/menu/events/main.js]: Just double checked and I had to do this in order to get some tests working but that didn't turn out to be the issue. Will remove it.

Comment (jac-ob) [reply lib/composites/menu/events/main.js]: I was wrong here, `useCapture` ensures that this fires first which is what we want

File: lib/commons/dialog/trap-focus/index.js [hunk 9]
```diff
@@ -37,4 +38,20 @@ module.exports = (selector) => {
       first.focus();
     }
   });
+
+
+  /**
+  * Prevents shift + tab from leaving alert
+  */
+  delegate(document.body, '.dqpl-alert .dqpl-dialog-inner, .dqpl-modal-header h2', 'keydown', (e) => {
```
Comment (schne324) [inline lib/commons/dialog/trap-focus/index.js]: test(s) for this?

File: lib/commons/animation/animation.less [hunk 10]
```diff
@@ -28,6 +28,12 @@
   .animate(opacity 400ms);
 }
 
+@{prefix}fadein-flex {
```
Comment (schne324) [inline lib/commons/animation/animation.less]: resolves #40

**Pull request merged**